src/main/java/org/openapitools/client/model/ModuleCondensed.java

```diff
@@ -175,22 +174,22 @@ private String toIndentedString(Object o) {
      * Gets or Sets semesters
      */
     public enum SemestersEnum {
-        NUMBER_1(new BigDecimal("1")),
+        S1("S1"),
 
-        NUMBER_2(new BigDecimal("2")),
+        S2("S2"),
 
-        NUMBER_3(new BigDecimal("3")),
+        ST1("ST1"),
 
-        NUMBER_4(new BigDecimal("4"));
+        ST2("ST2");
 
-        private BigDecimal value;
+        private String value;
 
-        SemestersEnum(BigDecimal value) {
+        SemestersEnum(String value) {
             this.value = value;
         }
 
         @JsonCreator
-        public static SemestersEnum fromValue(BigDecimal value) {
+        public static SemestersEnum fromValue(String value) {
```

Comment:
`fromValue` method will check if the String input is valid, will throw an `IllegalArgumentException` if the input is invalid. The method is overloaded such that it can also take in to check input that is of type Integer since semester in the semester data are integer 1-4.